stglib/aqd/aqdutils.py

```diff
@@ -1420,3 +1435,30 @@ def apply_wave_coord_output(ds, T, T_orig):
         ds = ds.drop(["VEL1", "VEL2", "VEL3"])
 
     return ds
+
+
+def fill_agc(ds):
+    """
+    Fill velocity data with AGC threshold.
+    Beam AGC variables (AGC1_1221, AGC2_1222, and AGC3_1223) are used to fill corresponding beam velocities (vel1_1277, vel2_1278, vel3_1279).
+    Average AGC (AGC_1202) is used to fill transformed eastward, northward, and upward velocities (u_1205, v_1206, w_1204).
+    """
+    if "agc_threshold" in ds.attrs:
```

Comment:
It's not clear if the threshold is a max or a min. We should probably support both. I suggest something like `velocity_agc_min` or `velocity_agc_max`.

Reply:
Edited to support both agc min and max under `fill_agc`. I can separate these in two different functions but thought keeping them together would be more concise.